### PR TITLE
Make viewport see-through flags more granular

### LINKED ISSUE #1180: Separate road and tracks visibility toggle.
**Describe the solution you'd like**
It would be nice to have a separate toggle for railroad tracks visibility and roads visibility.

**Describe alternatives you've considered**
Sometimes rotating and moving the camera around helps but does not solve the issue when there are roads and tracks close together at different height levels.

**Additional context**
Currently railroad tracks and roads hidden/visible toggle applies to both types at the same time, this makes creating dense networks a bit tricky sometimes.

**Screenshot 1: visibility on**
tracks_on
**Screenshot 2: visibility off**
tracks_off

### PR DESCRIPTION
**Pull request by AaronVanGeffen**
The way hiding viewport elements works in Locomotion has often been suggested for improvement (e.g. #116, #1180). Our codebase is now in such a state where this is up to be reworked. This PR refines the granularity of the see-through flags, and adds new shortcuts for them. I've opted to re-assign keys 3-9 to keep them consistent with the expanded viewport menu.

Aside from the refined granularity, the see-through flags now apply to the entire viewport, rather than only the bottom half. In #116, it was suggested to add a slider to configure the amount, however I personally think that is overly complicating it. I see more use in implementing a viewport clipper similar OpenRCT2's cut-away view. Something for another PR, for sure.

Closes #1180

### CODE REVIEW

**Review by LeftofZen — CHANGES_REQUESTED** [review]
Works very well, but found a tiny bug: the options for height marks are backwards:

image

image

**Comment by LeftofZen** [comment]
Any chance of an option for see-through vehicles? Or does that require a lot more code?

File: src/OpenLoco/src/Input/Shortcuts.h [hunk 1]
```diff
@@ -15,10 +15,13 @@ namespace OpenLoco::Input
...
-        toggleHideForegroundTracks,
-        toggleHideForegroundScenery,
-        toggleHeightMarksonLand,
-        toggleHeightMarksonTracks,
+        toggleSeeThroughTracks,
+        toggleSeeThroughRoads,
+        toggleSeeThroughTrees,
+        toggleSeeThroughBuildings,
+        toggleSeeThroughScenery,
+        toggleHeightMarksOnLand,
```
Comment (LeftofZen) [inline src/OpenLoco/src/Input/Shortcuts.h:23]: tracks should be before land

**Comment by AaronVanGeffen** [comment]
> Any chance of an option for see-through vehicles? Or does that require a lot more code?

Those have to do with painting entities rather than tile elements. I agree that would be useful, though. Could you file a feature request?

**Review by LeftofZen — APPROVED** [review]

**Pull request merged**